### PR TITLE
#126 Add the "question/suggestions" endpoint to the API

### PR DESCRIPTION
**Pull request by cesartg**

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -41,7 +41,8 @@ All api URIs start with `/api` followed by one of the below.
 | `/questions/{id}/user` | Show owner of question with {id} | No | No | `id` |
 | `/questions/search` | Search questions using ?term={term} param| Yes | Yes | `term` |
 | `/questions/unanswered` | Look for all unanswered questions| Yes | Yes | - |
-| `/questions/newest` | Look for all questions showing newest first| Yes | createdAt,desc + user choices | - |
+| `/questions/newest` | Look for all questions showing newest first| No | createdAt,desc + user choices | - |
```
Comment (wdonet) [inline API.md]: why was this changed to 'No' on pagination ? I'm still seeing the code supporting pageable param

File: src/main/java/com/nearsoft/questions/controller/api/QuestionApiController.java [hunk 2]
```diff
@@ -85,4 +91,26 @@ public QuestionApiController(QuestionSearchRepository searchRepository, Question
         });
         return new ResponseEntity<>(assembler.toResource(results), HttpStatus.OK);
     }
+
+    private static final int MAX_NUMBER_OF_SUGGESTIONS = 10;
```
Comment (wdonet) [inline src/main/java/com/nearsoft/questions/controller/api/QuestionApiController.java]: can this be moved to a configuration and have the constant as a default value just in case the configuration doesn't exist or has a null/empty value?

File: API.md (hunk 1 above)
Comment (cesartg) [inline API.md]: Unintended change. Reverted.

File: src/main/java/com/nearsoft/questions/controller/api/QuestionApiController.java (hunk 2 above)
Comment (cesartg) [inline src/main/java/com/nearsoft/questions/controller/api/QuestionApiController.java]: Configuration.java only exists in hackday branch. 
Should I merge this to the hackday branch?

Comment (wdonet) [inline src/main/java/com/nearsoft/questions/controller/api/QuestionApiController.java]: that's right, all development should be in hackday branch for now.  Your branch should derive from hackday.  Sorry for not making it clear before you started.

**Comment by cesartg** [comment]
Closing this. "issue-126-api-suggestions" doesn't derive from hackday branch.
